substrate/frame/support/src/storage/storage_noop_guard.rs

```diff
@@ -111,4 +135,34 @@ mod tests {
 			panic!("Something else");
 		});
 	}
+
+	#[test]
+	#[should_panic(expected = "StorageNoopGuard found unexpected storage changes.")]
+	fn storage_noop_guard_panics_created_from_error_message() {
+		TestExternalities::default().execute_with(|| {
+			let _guard = StorageNoopGuard::from_error_message(
+				"StorageNoopGuard found unexpected storage changes.",
+			);
+			frame_support::storage::unhashed::put(b"key", b"value");
+		});
+	}
+
+	#[test]
+	#[should_panic(expected = "StorageNoopGuard found unexpected storage changes.")]
+	fn storage_noop_guard_panics_with_set_error_message() {
+		TestExternalities::default().execute_with(|| {
+			let mut guard = StorageNoopGuard::default();
+			guard.set_error_message("StorageNoopGuard found unexpected storage changes.");
+			frame_support::storage::unhashed::put(b"key", b"value");
+		});
+	}
+
+	#[test]
+	#[should_panic(expected = "StorageNoopGuard detected wrongful storage changes.")]
+	fn storage_noop_guard_panics_new_alias() {
```

Comment:
do you mean something like
```suggestion
	fn storage_noop_guard_panics_with_default_message() {
```

Reply:
It's just a test added in to prevent regressions in the behaviour of the new() function if anybody were to make changes. Basically a quick test to check that it equates to default without implementing PartialEq.
A bit contrived I agree.